src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -116,6 +121,10 @@ private String setPlayerTeamToTeleporterTeam(EntityRef player, EntityRef telepor
     private void handlePlayerTeleport(EntityRef player, String team) {
         Vector3f randomVector = new Vector3f(-1 + random.nextInt(3), 0, -1 + random.nextInt(3));
         player.send(new CharacterTeleportEvent(randomVector.add(LASUtils.getTeleportDestination(team))));
+        Optional<Prefab> prefab = Assets.getPrefab("inventory");
+        StartingInventoryComponent startingInventory = prefab.get().getComponent(StartingInventoryComponent.class);
+        player.addComponent(startingInventory);
+        player.send(new RequestInventoryEvent());
```

Comment:
Having the same code twice indicates that you probably should refactor this. You might want to consider introducing a LaS-specific event denoting a "player reset", and then fire that event both when the player teleports into the game and when they respawn.

Reply:
As we will be initializing the startingInventory in the beginning of the systems and the inventories should eventually be different for spawn and respawn that would leave only the last 2 lines to be refactored which might be unnecessary.